site/src/pages/UserSettingsPage/NotificationsPage/NotificationsPage.tsx

```diff
@@ -0,0 +1,238 @@
+import {
+  notificationDispatchMethods,
+  selectTemplatesByGroup,
+  systemNotificationTemplates,
+  updateUserNotificationPreferences,
+  userNotificationPreferences,
+} from "api/queries/notifications";
+import type {
+  NotificationPreference,
+  NotificationTemplate,
+} from "api/typesGenerated";
+import { displaySuccess } from "components/GlobalSnackbar/utils";
+import { Loader } from "components/Loader/Loader";
+import { Stack } from "components/Stack/Stack";
+import { useAuthenticated } from "contexts/auth/RequireAuth";
+import {
+  castNotificationMethod,
+  methodIcons,
+  methodLabels,
+} from "modules/notifications/utils";
+import { pageTitle } from "utils/page";
+import { Section } from "../Section";
+
+type PreferenceSwitchProps = {
+  id: string;
+  disabled: boolean;
+  onToggle: (checked: boolean) => Record<string, boolean>;
+};
+
+const PreferenceSwitch: FC<PreferenceSwitchProps> = ({
+  id,
+  disabled,
+  onToggle,
+}) => {
+  const { user } = useAuthenticated();
+  const queryClient = useQueryClient();
+  const updatePreferences = useMutation(
+    updateUserNotificationPreferences(user.id, queryClient),
+  );
+
+  return (
+    <Switch
+      id={id}
+      size="small"
+      checked={!disabled}
+      onChange={async (_, checked) => {
+        await updatePreferences.mutateAsync({
+          template_disabled_map: onToggle(checked),
+        });
+        displaySuccess("Notification preferences updated");
+      }}
+    />
+  );
+};
+
+export const NotificationsPage: FC = () => {
+  const { user, permissions } = useAuthenticated();
+  const [disabledPreferences, templatesByGroup, dispatchMethods] = useQueries({
+    queries: [
+      {
+        ...userNotificationPreferences(user.id),
+        select: selectDisabledPreferences,
+      },
+      {
+        ...systemNotificationTemplates(),
+        select: (data: NotificationTemplate[]) => {
+          const groups = selectTemplatesByGroup(data);
+          return permissions.viewDeploymentValues
+            ? groups
+            : {
+                // Members only have access to the "Workspace Notifications" group
+                ["Workspace Events"]: groups["Workspace Events"],
+              };
+        },
+      },
+      notificationDispatchMethods(),
+    ],
+  });
+  const ready =
+    disabledPreferences.data && templatesByGroup.data && dispatchMethods.data;
+
+  return (
+    <>
+      <Helmet>
+        <title>{pageTitle("Notifications Settings")}</title>
+      </Helmet>
+      <Section
+        title="Notifications"
+        description="Configure your notification preferences. Icons on the right of each notification indicate delivery method, either SMTP or Webhook."
+        layout="fluid"
+      >
+        {ready ? (
+          <Stack spacing={3}>
+            {Object.entries(templatesByGroup.data).map(([group, templates]) => {
+              const allDisabled = templates.some((tpl) => {
```

Comment:
Related to the first comment, I see how this can be confusing. I will put more thought into it. I just would rather do it in a different PR to avoid this one getting longer than it is.

Reply:
> I also think the current email icons are a bit misleading, because before I tried interacting with them, I expected them to be buttons. As in, clicking one would send off an email in some way

I can try a badge and see how it looks like

Reply:
Yeah, I can deal with a slightly confusing UI (or at least confusing from my point of view) as long as it works predictably